eureka2-core/src/main/java/com/netflix/eureka2/utils/rx/RetryStrategyFunc.java

```diff
@@ -0,0 +1,48 @@
+package com.netflix.eureka2.utils.rx;
+
+import rx.Observable;
+import rx.functions.Func1;
+import rx.functions.Func2;
+
+import java.util.concurrent.TimeUnit;
+
+/**
+ * A Func1 with retry options for use in .retryWhen(new RetryStrategyFunc())
+ *
+ * @author David Liu
+ */
+public class RetryStrategyFunc implements Func1<Observable<? extends Throwable>, Observable<Long>> {
```

Comment:
Is it visible in the API? If not, depending on ocelli is ok.

On Fri, Jan 16, 2015 at 11:35 AM, qiangdavidliu <notifications@github.com>
wrote:

> In
> eureka2-core/src/main/java/com/netflix/eureka2/utils/rx/RetryStrategyFunc.java
> <https://github.com/Netflix/eureka/pull/347#discussion-diff-23103609>:
>
> > @@ -0,0 +1,48 @@
> > +package com.netflix.eureka2.utils.rx;
> > +
> > +import rx.Observable;
> > +import rx.functions.Func1;
> > +import rx.functions.Func2;
> > +
> > +import java.util.concurrent.TimeUnit;
> > +
> > +/**
> > + * A Func1 with retry options for use in .retryWhen(new RetryStrategyFunc())
> > + *
> > + * @author David Liu
> > + */
> > +public class RetryStrategyFunc implements Func1<Observable<? extends Throwable>, Observable<Long>> {
>
> The Ocelli version is nicer, but we don't want to have dependency on
> ocelli for all the packages that needs retry. It feels like this is
> something useful to put in a more basic layer (rxjava/rxnetty)
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/Netflix/eureka/pull/347/files#r23103609>.
>